BuildJson.java

```diff
@@ -0,0 +1,69 @@
+import static my.Constants.*;
+
+import org.json.*;
+import java.util.UUID;
+
+public class BuildJson {
+
+  public JSONObject buildJson(String commands[]) throws JSONException{
```

Comment:
This approach will work, however it's hard to conceptualize the hierarchy of request and response because it's implemented via encoding plain fields directly to json objects and vice versa. 

It would be much easier to read if there were data structures present, e.g.

```Java
public class Request implements Serializable {
 
    private static final long serialVersionUID = 1L;
     
    private Integer id;
    private String Command;
 
    //Setters and Getters
}
```

(I took the example from this location:  https://howtodoinjava.com/jaxb/convert-object-to-json-moxy/)

This approach will also help you to focus on the structure of the request/response and the serialization layer will be secondary to the protocol/structures passed to the methods.

Let me know if this makes sense.

Reply:
I have pushed changes to the branch. 

- Added a request and response data structure
- Modified BuildJson to build requests and responses using those data structures
- Modified BuildJson to no longer create notification data structures 
- Modified ParseJson to expect a request or response data structure

Reply:
@moyaog now, when you are using `Serializable` I think you don't really need to use manual `BuildJson` approach, you can simply use `AXBContext.newInstance(Request.class);` as listed here in 2.3:

https://howtodoinjava.com/jaxb/convert-object-to-json-moxy/

Let me know if this makes sense

Reply:
Are you suggesting just using the` JAXBContext.newInstance(Request.class);` in lieu of the BuildJson class? If that is the case, I would just be concerned that that doesn't follow the JSON-RPC specification.

Reply:
@moyaog makes sense then, let's keep it as is to conform to JSON-RPC spec

Reply:
Marshal/unmarshal layer lgtm, let's move on to other parts